toffy/mph_comp.py

```diff
@@ -0,0 +1,162 @@
+    io_utils.validate_paths(bin_file_path)
+
+    # get fov json file in bin_file_path
+    json_file = io_utils.list_files(bin_file_path, fov+".json")
+    if len(json_file) == 0:
+        raise FileNotFoundError(f"The FOV name supplied doesn't have a JSON file: {fov}")
+
+    # retrieve estimated time (frame dimensions x pixel dwell time)
+    with open(os.path.join(bin_file_path, json_file[0])) as file:
+        run_metadata = json.load(file)
+        try:
+            size = run_metadata.get('frameSize')
+            time = run_metadata.get('dwellTimeMillis')
+            estimated_time = int(size**2 * time)
+        except TypeError:
+            raise KeyError("The FOV json file is missing one of the necessary keys "
+                           "(frameSize or dwellTimeMillis)")
+
+    return estimated_time
+
+
+def compute_mph_metrics(bin_file_path, csv_dir, fov, target, mass_start, mass_stop):
+    """Retrieves FOV total counts, pulse heights, & estimated time for all fovs in the directory
+        Args:
+            bin_file_path (str): path to the FOV bin and json files
+            csv_dir (str): path to output csv to
+            fov (string): name of fov bin file without the extension
+            target (str): channel to use
+            mass_start (float): beginning of mass integration range
+            mass_stop (float): end of mass integration range
+
+            """
+
+    # path validation
+    io_utils.validate_paths(bin_file_path)
+    io_utils.validate_paths(csv_dir)
+
+    # retrieve the data from bin file and output to individual csv
+    pulse_height_file = fov + '-pulse_height.csv'
+
+    try:
+        median = bin_files.get_median_pulse_height(bin_file_path, fov,
+                                                   target, (mass_start, mass_stop))
+        count_dict = bin_files.get_total_counts(bin_file_path, [fov])
+    except FileNotFoundError:
+        raise FileNotFoundError(f"The FOV name supplied doesn't have a JSON file: {fov}")
+    except ValueError:
+        raise ValueError(f"The target name is invalid: {target}")
+
+    count = count_dict[fov]
+    time = get_estimated_time(bin_file_path, fov)
+
+    out_df = pd.DataFrame({
+        'fov': [fov],
+        'MPH': [median],
+        'total_count': [count],
+        'time': [time]})
+
+    # saves individual .csv  files to bin_file_path
+    if not os.path.exists(os.path.join(csv_dir, pulse_height_file)):
+        out_df.to_csv(os.path.join(csv_dir, pulse_height_file), index=False)
+
+
+def combine_mph_metrics(csv_dir, return_data=False):
+    """Combines data from individual csvs into one
+        Args:
+            csv_dir (str): path where FOV csvs are stored
+            return_data (bool): whether to return dataframe with mph metrics, default False
+
+        Returns:
+            combined mph data for all FOVs
+            """
+
+    # path validation checks
+    io_utils.validate_paths(csv_dir)
+
+    # list bin files in directory
```

Comment:
Having inaccurate comments is worse than having no comments. Make sure if you change your code that you change the comments as well